src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double ARRAY_MULTIPLIER = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("There is no such index");
+        }
+        grow();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
```

Comment:
A hint: if you want to iterate over ALL elements in array/collection (without using any specific starting/ending index), please use `forEach` loop.

Reply:
I tried to use forEach but I am getting this information: "Foreach not applicable to type 'core.basesyntax.List<T>'"

Reply:
![2024-04-24](https://github.com/mate-academy/jv-array-list/assets/128905637/7bd8c742-84c6-413b-bbf1-73b4b07597ae)

Reply:
My bad - it will work only for `Iterable`s.
Let's keep it that way.